tests/drivers/fuel_gauge/sy24561/testcase.yaml

```diff
@@ -5,3 +5,18 @@ tests:
     filter: dt_compat_enabled("silergy,sy24561")
     platform_allow:
       - native_sim
+      - native_sim/native/64
+    harness: console
+    harness_config:
+      type: multi_line
+      ordered: true
+      regex:
+        - "Property 65535 not supported"
+        - "SY24561: SOC alarm threshold 0 should be >= 1"
+        - "emul_sy24561_reg_write: 0x0c 0x3c1f"
+        - "SY24561: SOC alarm threshold 255 should be <= 32"
+        - "emul_sy24561_reg_write: 0x0c 0x3c00"
+        - "SY24561: Temperature 0.0K should be >= 253.1K"
+        - "emul_sy24561_reg_write: 0x0c 0x141c"
+        - "SY24561: Temperature 6553.5K should be <= 333.1K"
+        - "emul_sy24561_reg_write: 0x0c 0x641c"
```

Comment:
I have not validated these values against the datasheet nor hardware. Still, let's check against the commited driver version which was tested at the contribution time